build.rs

```diff
@@ -0,0 +1,159 @@
+use bzip2::read::BzDecoder;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
+use tar::Archive;
+use zip::ZipArchive;
+
+fn main() {
+    // Tell Cargo to rerun the build script if the build script itself changes.
+    println!("cargo:rerun-if-changed=build.rs");
+
+    let testgen_lib_version = "10.1.2.1";
+
+    let target_os = if cfg!(target_os = "macos") {
+        "darwin"
+    } else if cfg!(target_os = "linux") {
+        "linux"
+    } else if cfg!(target_os = "windows") {
+        "windows"
+    } else {
+        panic!("Unsupported OS");
+    };
+
+    let arch = if cfg!(target_arch = "x86_64") {
+        "x86_64"
+    } else if cfg!(target_arch = "aarch64") {
+        "aarch64"
+    } else {
+        panic!("Unsupported architecture");
+    };
+
+    let suffix = if target_os == "windows" {
+        ".zip"
+    } else {
+        ".tar.bz2"
+    };
+
+    let file_name = format!("testgen-hs-{}-{}-{}", testgen_lib_version, arch, target_os);
+    let download_url = format!(
+        "https://github.com/input-output-hk/testgen-hs/releases/download/{}/{}{}",
+        testgen_lib_version, file_name, suffix
+    );
+
+    println!("Fetching {}", file_name);
+
+    // Cache directory for testgen-hs artifacts.
+    let cache_dir = dirs::cache_dir()
+        .expect("Unable to determine cache directory")
+        .join("testgen-hs")
+        .join(testgen_lib_version);
```

Comment:
So the final directory is:

```
~/.cache/testgen-hs/10.1.2.1/extracted/testgen-hs
```

What would you say to maybe adding one more segment, `blockfrost-platform`, i.e.

```
~/.cache/blockfrost-platform/testgen-hs/10.1.2.1/extracted/testgen-hs
```

If anything comes up later, we’ll only ever have a single directory in `~/.cache` :)

-----

OTOH, since this is a _build_ script, I think it should write to the Cargo's `target/` directory of the project? Like where it puts other binaries/libs:

```
❯ cd ~/Work/blockfrost-platform
❯ ls -lh target/debug/deps/libserde_with_macros-1ef82c761f38d70a.so
-rwx------ 1 mw users 3.9M Jan 14 14:23 target/debug/deps/libserde_with_macros-1ef82c761f38d70a.so
```

Reply:
Thanks!